src/test/java/com/olgaboiar/mint/ServerTest.java

```diff
@@ -38,15 +38,15 @@ void testServerIsReachableAfterStarting () throws Exception {
         assertTrue(reachable);
     }
 
-    @Test
```

Comment:
Injecting the dependencies rather than having your main server class instantiate them can make testing easier combined with interfaces and mocking.